etw/eventdata.go

```diff
@@ -0,0 +1,23 @@
+package etw
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+// EventData maintains a buffer which builds up the data for an ETW event. It
+// needs to be paired with EventMetadata which describes the event.
+type EventData struct {
+	buffer bytes.Buffer
+}
+
+// NewEventData returns a new EventData with an empty buffer.
+func NewEventData() *EventData {
+	return &EventData{}
+}
+
+// AddString appends the data for a string to the end of the buffer.
+func (ed *EventData) AddString(data string) {

```

Comment:
Do you plan on adding an overload for all types? Should we just use `AddItem(data interface{})`

Reply:
It will depend on the specifics of other types we want to support, and how tracelogging requires them to be packed. I don't think we can support a generic AddItem() at this level, but rather at a higher level which calls into these functions to do the work. Right now i"m planning to add more here as needed while testing this out.

As an example of another way to pack data: tracelogging supports counted strings, which are prefixed with length.

Reply:
For a lot of blittable types like numbers we should be able to write their data using a single function.

Reply:
Should we just add a function for each intrinsic type at least then?

Reply:
I'll expand on this in a future PR.